File: samplestore/src/main/java/com/stripe/samplestore/service/StripeService.java
package com.stripe.samplestore.service;

import java.util.Map;

import io.reactivex.Observable;
import okhttp3.ResponseBody;
import retrofit2.http.Body;
import retrofit2.http.FieldMap;
import retrofit2.http.FormUrlEncoded;
import retrofit2.http.POST;

/**
 * The {@link retrofit2.Retrofit} interface that creates our API service.
 */
public interface StripeService {

    @POST("capture_payment")
    Observable<ResponseBody> capturePayment(@Body Map<String, Object> params);

    /**
     * Used for Payment Intent Manual confirmation
     *
     * @see <a
     * href="https://stripe.com/docs/payments/payment-intents/quickstart#manual-confirmation-flow">
     * Manual Confirmation Flow</a>
     */
    @POST("confirm_payment")
    Observable<ResponseBody> confirmPayment(@Body Map<String, Object> params);

    @FormUrlEncoded
    @POST("ephemeral_keys")
    Observable<ResponseBody> createEphemeralKey(@FieldMap Map<String, String> apiVersionMap);
}

File: samplestore/src/main/java/com/stripe/samplestore/service/StripeService.kt
package com.stripe.samplestore.service

import io.reactivex.Observable
import okhttp3.ResponseBody
import retrofit2.http.Body
import retrofit2.http.FieldMap
import retrofit2.http.FormUrlEncoded
import retrofit2.http.POST

/**
 * The [retrofit2.Retrofit] interface that creates our API service.
 */
interface StripeService {

    /**
     * Returns the PaymentIntent client secret in the format shown below.
     *
     * {"secret": "pi_1Eu5SqCRMb_secret_O2Avhk5V0Pjeo"}
     */
    @POST("capture_payment")
    fun capturePayment(@Body params: HashMap<String, Any>): Observable<ResponseBody>

    /**
     * Used for Payment Intent Manual confirmation
     *
     * @see [Manual Confirmation Flow](https://stripe.com/docs/payments/payment-intents/quickstart.manual-confirmation-flow)
     *
     * Returns the PaymentIntent client secret in the format shown below.
     *
     * {"secret": "pi_1Eu5SqCRMb_secret_O2Avhk5V0Pjeo"}
     */
    @POST("confirm_payment")
    fun confirmPayment(@Body params: HashMap<String, Any>): Observable<ResponseBody>

    @POST("create_setup_intent")
    fun createSetupIntent(@Body params: HashMap<String, Any>): Observable<ResponseBody>

    @FormUrlEncoded
    @POST("ephemeral_keys")
    fun createEphemeralKey(@FieldMap apiVersionMap: HashMap<String, String>): Observable<ResponseBody>
}
